engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreHeartbeatWriteActor.scala

```diff
@@ -40,21 +40,37 @@ case class WorkflowStoreHeartbeatWriteActor(workflowStoreAccess: WorkflowStoreAc
     *
     * @return the number of elements processed
     */
-  override protected def process(data: NonEmptyVector[(WorkflowId, OffsetDateTime)]): Future[Int] = instrumentedProcess {
-    val heartbeatDateTime = OffsetDateTime.now()
-    val processFuture = workflowStoreAccess.writeWorkflowHeartbeats(data, heartbeatDateTime)
-    processFuture transform {
-      // Track the `Try`, and then return the original `Try`. Similar to `andThen` but doesn't swallow exceptions.
-      _ <| trackRepeatedFailures(heartbeatDateTime, data.length)
+  override protected def process(data: NonEmptyVector[WorkflowStoreWriteHeartbeatCommand]): Future[Int] = instrumentedProcess {
+    val now = OffsetDateTime.now()
+    val workflowsIdsHavingStaleHeartbeats: Seq[WorkflowId] = data.collect {
+      case h if JDuration.between(h.heartbeatTime, now).toNanos >= failureShutdownDuration.toNanos => h.workflowId
+    }
+
+    if (workflowsIdsHavingStaleHeartbeats.isEmpty) {
+      val processFuture = workflowStoreAccess.writeWorkflowHeartbeats(data.map { h => (h.workflowId, h.submissionTime) }, now)
```

Comment:
TOL is it worth logging a warning if the difference is significantly bigger than the usual heartbeat interval, but less than the shutdown timeout?

TOL2: Or even better, is it worth keeping a metric of these intervals over time?

Reply:
warning added